### ISSUE TITLE
"Subtract fee from amount" not correctly calculating send amount

### ISSUE THREAD

**Issue by belovachap** [peercoin/peercoin#645]
**Expected behavior**

![image](https://user-images.githubusercontent.com/396781/200140218-4de46c48-fcdf-42c4-b54f-e3b7ef51be9d.png)

As you can see in the screen shot, I'm trying to send 1.0 PPC with the fee subtracted from it. Typically this means only 1.0 PPC in total is consumed. 

**Actual behavior**

But for some reason it keeps trying to send 1.001548 PPC :thinking: If the fee for the transaction is 0.007958 PPC then it should be sending 0.992042 PPC for a total amount of 1.0 PPC.

![image](https://user-images.githubusercontent.com/396781/200140571-d649984e-0f00-4b79-97b5-f6f024f2d7e9.png)

**To reproduce**

Subtract fee from amount has worked flawlessly for me up until this one send. It's also working again after doing this one send so I'm really not sure how to reproduce :(

**System information**

v0.11.8, Ubuntu 22.04

**Comment by belovachap** [comment]
Lightning strikes twice!

![image](https://user-images.githubusercontent.com/396781/200181070-39e6b146-f8b7-44d0-8c94-5e14f909a1bb.png)

I restarted the client with the `-debug` flag and saw this output when I made the transaction:

```
2022-11-06T16:04:39Z GUI: NotifyKeyStoreStatusChanged
2022-11-06T16:04:39Z [default wallet] keypool added 1 keys (0 internal), size=1000 (0 internal)
2022-11-06T16:04:39Z [default wallet] keypool reserve 5129
2022-11-06T16:04:39Z [default wallet] keypool keep 5129
2022-11-06T16:04:39Z Fee Calculation: Fee:12467 Bytes:1082 Needed:10820
```

The `Needed:10820` seems to be what's subtracted from the send amount but the `Fee:12467` is what actually gets added on?

**Comment by ihavenoface** [comment]
Since you were using rather small inputs for that transaction, I believe you're dealing with [this rule here](https://github.com/peercoin/peercoin/blob/64b8125/src/wallet/wallet.cpp#L2979-L2984):
```
// peercoin: sub-cent change is moved to fee
if (nChange > 0 && nChange < MIN_TXOUT_AMOUNT)
{
    nFeeRet += nChange;
    nChange = 0;
}
```
The wallet is trying to get rid of smaller balances by using them to fill up the requested amount + fee. Since the remainder would end up being dust, it's being dropped instead of pushing it to a new txout. Same old reasoning it drops fees in the first place: to reduce chainweight.

If you were to select a bigger input of say 5PPC this wouldn't occur.

**Comment by belovachap** [comment]
@ihavenoface seems like you might be right about what is happening. I'm still kind of confused about why there's both a `Fee` and `Needed` amount calculated since it seems like what we want to be using is only the `Fee` in this case.

**Comment by ihavenoface** [comment]
Including both gave you a clue on how the inputs are handled exactly would be one explanation.

**Comment by backpacker69** [comment]
fixed in release-0.12
